packages/mui-joy/src/Radio/Radio.tsx

```diff
@@ -302,9 +302,10 @@ const Radio = React.forwardRef(function Radio(inProps, ref) {
   const name = inProps.name || radioGroup?.name || nameProp;
   const disableIcon = inProps.disableIcon || radioGroup?.disableIcon || disableIconProp;
   const overlay = inProps.overlay || radioGroup?.overlay || overlayProp;
+  const isValueValid = value !== undefined && value !== null && value !== '';
 
   const radioChecked =
-    typeof checkedProp === 'undefined' && !!value
+    typeof checkedProp === 'undefined' && isValueValid
```

Comment:
After some research, I found that in most cases, an empty string is considered a valid value. If you pass an empty string to a radio group or a radio input, it won't be selected. I apologize for the previous comment. We can revert to previous commit or do:
```suggestion
    typeof checkedProp === 'undefined' && value != null
```